manifests/charts/base/templates/crds.yaml

```diff
@@ -1,3 +1,58 @@
-{{- if .Values.base.enableCRDTemplates }}
-{{ .Files.Get "crds/crd-all.gen.yaml" }}
-{{- end }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: extended-crds
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded 
+data:
+  extended.gen.yaml: |
+    {{ .Files.Get "files/extended.gen.yaml" | nindent 10 }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: stable-crds
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded 
+data:
+  stable.gen.yaml: |
+    {{ .Files.Get "files/stable.gen.yaml" | nindent 10 }}
+---
+apiVersion: batch/v1
```

Comment:
I don't think using a job to install CRDs is acceptable. This will break all sorts of things, and introduces a ton of  operational burden. It won't work for many install mechanisms either

Reply:
Do you know of a better way to template the installation of the target crd file? 

I considered keeping the helm 3  approach of placing CRDs in the `crds` folder. Where `extended.gen.yaml`, the default channel, is the only one in `crds` folder. 

And users can manually switch to the stable channel by customizing their istio install to swap the files. Or we can design a tool to facilitate this. Alternatively we can ship Istio Stable and Istio Extended separately.

Reply:
Users shouldn't have to swap the files; that's too much toil. One of the helm 3 options besides the CRD folder is to put the CRDs in their own chart (which is basically what the base chart is). Just keep them flat, outside of the CRDs folder

Reply:
@keithmattix I previously had the helm chart conditionally apply the CRDs but I ran into issues in the helm upgrade workflow, requiring additional annotations, to tell `helm upgrade` to skip/include upgrading the CRDs. 

As I now know, pre-install jobs are worst, I will revisit adding the additional annotations, etc. to get this working as expected when helm chart conditionally installs the CRDs.